libafl/src/executors/inprocess.rs

```diff
@@ -692,63 +715,80 @@ mod windows_exception_handler {
-                .objective_mut()
-                .is_interesting(state, event_mgr, input, observers, &ExitKind::Timeout)
-                .expect("In timeout handler objective failure.");
+                let input = (data.timeout_input_ptr as *const I).as_ref().unwrap();
+                data.timeout_input_ptr = ptr::null_mut();
 
-            if interesting {
-                let mut new_testcase = Testcase::new(input.clone());
-                new_testcase.add_metadata(ExitKind::Timeout);
-                fuzzer
+                let interesting = fuzzer
                     .objective_mut()
-                    .append_metadata(state, &mut new_testcase)
-                    .expect("Failed adding metadata");
-                state
-                    .solutions_mut()
-                    .add(new_testcase)
-                    .expect("In timeout handler solutions failure.");
-                event_mgr
-                    .fire(
-                        state,
-                        Event::Objective {
-                            objective_size: state.solutions().count(),
-                        },
-                    )
-                    .expect("Could not send timeouting input");
-            }
+                    .is_interesting(state, event_mgr, input, observers, &ExitKind::Timeout)
+                    .expect("In timeout handler objective failure.");
+
+                if interesting {
+                    let mut new_testcase = Testcase::new(input.clone());
+                    new_testcase.add_metadata(ExitKind::Timeout);
+                    fuzzer
+                        .objective_mut()
+                        .append_metadata(state, &mut new_testcase)
+                        .expect("Failed adding metadata");
+                    state
+                        .solutions_mut()
+                        .add(new_testcase)
+                        .expect("In timeout handler solutions failure.");
+                    event_mgr
+                        .fire(
+                            state,
+                            Event::Objective {
+                                objective_size: state.solutions().count(),
+                            },
+                        )
+                        .expect("Could not send timeouting input");
+                }
 
-            event_mgr.on_restart(state).unwrap();
+                event_mgr.on_restart(state).unwrap();
 
-            #[cfg(feature = "std")]
-            println!("Waiting for broker...");
-            event_mgr.await_restart_safe();
-            #[cfg(feature = "std")]
-            println!("Bye!");
+                #[cfg(feature = "std")]
+                println!("Waiting for broker...");
+                event_mgr.await_restart_safe();
+                #[cfg(feature = "std")]
+                println!("Bye!");
 
-            event_mgr.await_restart_safe();
+                event_mgr.await_restart_safe();
+
+                LeaveCriticalSection(
```

Comment:
We should probably place a compiler barrier before this, we're using low level windows API here so Rust won't know it is not allowed to reorder, right?

Reply:
yes